src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -1243,6 +1248,46 @@ private void ParseModifiers(SyntaxListBuilder tokens, bool forAccessors)
                             break;
                         }
 
+                    case DeclarationModifiers.File:
+                        // 'file' is only a modifier if it is followed by an optional sequence of modifiers and then by a type keyword.
+                        {
+                            isFollowedByType ??= computeFollowedByType();
+                            if (isFollowedByType.GetValueOrDefault())
+                            {
+                                // this is a file modifier.
+                                modTok = ConvertToKeyword(EatToken());
+                                break;
+                            }
+
+                            return;
+
+                            bool computeFollowedByType()
+                            {
+                                var resetPoint = this.GetResetPoint();
+                                try
+                                {
+                                    while (true)
+                                    {
+                                        if (IsTypeDeclarationStart())
+                                        {
+                                            return true;
+                                        }
+                                        else if (!SyntaxFacts.IsKeywordKind(CurrentToken.ContextualKind))
```

Comment:
I don't understand this code. Why are we accepting random keywords here? I think we should be doing the same thing as we do for `async`: check to see if we should treat it as a type or a modifier, and parse appropriately. People are going to put `file` on things it shouldn't be on, such as members, local functions, and more.